ru/ru_gen.py

```diff
@@ -378,6 +390,14 @@ def simple_analysis(
 
             log_decision()
 
+
+        if len(unblock_batch_action_list)>0:
```

Comment:
We shouldn't need to check the length of these lists sending an empty list should be fine.